src/dto/ResponseStructures.py

```diff
@@ -33,6 +33,15 @@ class FilterResponse(BaseModel):
     result: Literal["YES", "NO"] = Field(
         description="'YES' if it matches a known false positive, otherwise 'NO'."
     )
+    filter_confidence: float = Field(
+        default=0.5,
```

Comment:
```suggestion
        default=None,
```

It can be misleading when the model fails to assign confidence but still gets 0.5. wdyt?

Also, is there a reason for the different defaults between `FilterResponse.filter_confidence` (`0.5`) and `AnalysisResponse.filter_confidence `(`None`)?